src/combat.cpp

```diff
@@ -1402,17 +1404,17 @@ void MagicField::onStepInField(Creature* creature)
 			bool harmfulField = true;
 
 			if (g_game.getWorldType() == WORLD_TYPE_NO_PVP || getTile()->hasFlag(TILESTATE_NOPVPZONE)) {
-				Creature* owner = g_game.getCreatureByID(ownerId);
+				auto owner = g_game.getCreatureByID(ownerId);
 				if (owner) {
 					if (owner->getPlayer() || (owner->isSummon() && owner->getMaster()->getPlayer())) {
 						harmfulField = false;
 					}
 				}
 			}
 
-			Player* targetPlayer = creature->getPlayer();
+			std::shared_ptr<Player> targetPlayer = creature->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
			const auto targetPlayer = creature->getPlayer();
```